### ISSUE TITLE
Seperate Ansible Plugin Integration tests from Role tests

### ISSUE THREAD

**Issue by ishanjainn** [grafana/grafana-ansible-collection#60]
Issue:
The CI Test(ci-test.yml GitHub Action) was initially build to test plugins in PRs quickly against a single linux architectures.
The integration-test.yml was instead build to do a complete test of plugins against various linux architectures.
Now that we have the grafana role(Agent role currently doesnt have any testing), The ci-test.yml takes a lot of time time complete (>40 mins). Instead it would be nice to seperate the plugin tests from role tests.

Incomplete Solution:
We can add a individual steps in action to run integration tests for particular plugin but that looks a bit manual. Is there a simpler way to separate the two tests(plugin vs role) 

Tagging @gardar if you know a simpler method for separating the grafana role tests or maybe speed up the grafana role test?

**Comment by ishanjainn** [comment]
https://github.com/grafana/grafana-ansible-collection/actions/runs/5348123271/jobs/9697568046
Plus although all steps run, It still fails the integration test for some reason

**Comment by gardar** [comment]
Perhaps you want to take inspiration from the ci workflow I made for the [prometheus ansible collection](https://github.com/prometheus-community/ansible). It's designed to run only the relevant tests on pull requests, it does this by looking at the files that have been modified.

If you like I can make a PR which implements this.
I have actually done some of the initial work in my fork which I used when testing the migration of the grafana server role to the collection as I didn't want to wait for the full test to run.
